public/app/plugins/datasource/cloudwatch/datasource.test.ts

```diff
@@ -1,88 +1,101 @@
-import { DataQueryResponse, dateTime, DefaultTimeRange } from '@grafana/data';
+import { of } from 'rxjs';
 import { setBackendSrv } from '@grafana/runtime';
+import { dateTime, DefaultTimeRange, observableTester } from '@grafana/data';
+
 import { TemplateSrv } from '../../../features/templating/template_srv';
 import { CloudWatchDatasource } from './datasource';
 
 describe('datasource', () => {
   describe('query', () => {
-    it('should return error if log query and log groups is not specified', async () => {
+    it('should return error if log query and log groups is not specified', done => {
       const { datasource } = setup();
-      const response: DataQueryResponse = (await datasource
-        .query({
+
+      observableTester().subscribeAndExpectOnNext({
```

Comment:
Sorry don't want to be nitpicky and it's not really relevant for this PR and we already talked about it a bit but I still don't understand the benefits of the `observableTester` vs just doing `toPromise()` or probably better approach than both of them with `toArray().toPromise()`.

Looking at the `observableTester` and if I read the code right it does not allow to check multiple emits or multiple errors so ti basically behaves the same maybe with the distinction that if there are multiple emits the `observableTester` will fail if the second emit does not pass the check, while `toPromise` will just ignore it. In this sense `toArray().toPromise` is probable best because you can check all the values after the observable completed and check whether there are more values than you would expect for example or test them one by one.

Also some of the `observableTester` API like `subscribeAndExpectOnNextAndComplete` has:
```
      next: (value: T) => {
        try {
          expectOnNext(value);
          done();
        } catch (err) {
          done.fail(err);
        }
      },
```
Which if I understand it will call jest `done` to end the test after the first emit but it does not unsubscribe from the observable or does it?

So I am not sure if I read the code right, if not I would like to know how it works.

Reply:
I understand @aocenas but I will remove all occurrences of observableTester and the observableTester and use the new jest matches introduced in #28929 instead. I'll do this in a separate PR. Thanks for pushing me in the right direction.

Reply:
Ah that matchers approach is really nice. Yeah that makes sense definitely no need to do it in this PR.